### PR TITLE
Fix BGZF parser failure:

### PR DESCRIPTION
**Pull request by allieychen**
- Fix the bug when discarding the first line and reading the last line for one block if the record is larger than one GZIP block.
- Use a variable to check whether it is the first time.

### CODE REVIEW

File: gcp_variant_transforms/beam_io/bgzf.py [hunk 1]
```diff
@@ -52,13 +52,19 @@ class BGZF(filesystem.CompressedFile):
   This class repeatedly fetches the `unused_data` from the decompressor until
   all GZIP blocks are decompressed.
   """
+  def __init__(self,
+               fileobj,
+               compression_type=filesystem.CompressionTypes.GZIP):
+    super(BGZF, self).__init__(fileobj, compression_type)
+    self._first_time = True
```
Comment (kemp-google) [inline gcp_variant_transforms/beam_io/bgzf.py]: _first_fetch ?

File: gcp_variant_transforms/beam_io/bgzf.py [hunk 2]
```diff
@@ -99,7 +105,7 @@ def _read_data_from_source(self):
     return self._file.read(self._read_size)
 
   def _is_first_time(self):
```
Comment (kemp-google) [inline gcp_variant_transforms/beam_io/bgzf.py]: is it really worth wrapping this field in a method?

File: gcp_variant_transforms/beam_io/bgzf.py [hunk 3]
```diff
@@ -153,13 +159,9 @@ class BGZFBlock(BGZF):
 
   It reads contents from `self._block`.
     - The string before first `\n` is discarded.
-    - Decompress the next 64KB after self._block and reads the first line. It
-      assumes one row can at most spans in two Blocks.
+    - Retrieves the next 16MB after self._block and reads the first line. It
```
Comment (kemp-google) [inline gcp_variant_transforms/beam_io/bgzf.py]: Why the change here from 64K to 16MB?

File: gcp_variant_transforms/beam_io/bgzf.py (hunk 1 above)
Comment (allieychen) [reply gcp_variant_transforms/beam_io/bgzf.py]: Done.

File: gcp_variant_transforms/beam_io/bgzf.py (hunk 2 above)
Comment (allieychen) [reply gcp_variant_transforms/beam_io/bgzf.py]: Done.

File: gcp_variant_transforms/beam_io/bgzf.py (hunk 3 above)
Comment (allieychen) [reply gcp_variant_transforms/beam_io/bgzf.py]: I have updated the doc to explain where the 16 MB comes from.The bug occurs when the record spans more than one GZIP block (max size 64 KB). 16 MB is the default read size for compressed files. I think it should be safe enough to cover one record (in compressed form).
